src/check.rs

```diff
@@ -0,0 +1,70 @@
+use std::path::PathBuf;
```

Comment:
`check` and `run` function have many common functionality, consider factoring it out